src/app/games/hive/hive.component.ts

```diff
@@ -80,47 +127,50 @@ export class HiveComponent extends HexagonalGameComponent<HiveRules, HiveMove, H
         this.pieces = [];
         for (const coord of this.getState().occupiedSpaces()) {
             const stack: HivePieceStack = this.getState().getAt(coord);
-            this.pieces.push({ coord, stack });
-        }
-        this.pieces.sort((a: StackAndCoord, b: StackAndCoord): number => {
-            if (a.coord.y === b.coord.y) return a.coord.x - b.coord.x;
-            else return a.coord.y - b.coord.y;
-        });
-        this.spaces = this.getAllNeighbors();
-        this.inspectedStack = MGPOptional.empty(); // TODOTODO: shouldn't cancelMoveAttempt be called before updateBoard?
+            const x: number = coord.x;
+            const y: number = coord.y;
+            for (let z: number = 0; z < stack.size(); z++) {
+                if (z in this.pieces === false) this.pieces[z] = new SpaceLevel();
```

Comment:
what ? this.pieces est de type "collection d'objet" et tu cherche si un nombre est une de ces cléf ? Si c'est pour checker si y a un z-ième élément c'p-e pas l'plus beau

Reply:
quoi que c'est p-e aussi assez classe... mmh, tu trouves ça classe ?

Reply:
faut pas abuser du `bli in blah` dans les tests, mais ici c'est le seul moyen me semble ?
À moins de faire un MGPMap, mais c'pas aussi facile à indexer du coup me semble